### PR TITLE
Kevin/introduce section cluster management to reference

### PR DESCRIPTION
**Pull request by kevinkq**
## Summary of the changes / Why this improves CrateDB

This introduces a new section cluster management to the reference. The document consists of a basic introduction to the crate-node command and its three modes ``repurpose``, ``unsafe-bootstrap``, and ``detach cluster``.  It originates from the following document: crate-node.cli under guide/docs/best-practices.

## Checklist
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes

### CODE REVIEW

File: blackbox/docs/admin/cluster-management.rst [hunk 1]
```diff
@@ -0,0 +1,55 @@
+.. _cluster_management:
+
+==================
+Cluster management
+==================
+
+With the ``crate-node`` command and the crate-node CLI tool, you can manage your
```
Comment (mechanomi) [inline blackbox/docs/admin/cluster-management.rst]: the crate-node command and CLI tool are one and the same

File: blackbox/docs/admin/cluster-management.rst [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+CrateDB clusters by forming new clusters, repairing broken clusters, and broken
+nodes.
+
+The ``crate-node`` command
+--------------------------
+
+The ``crate-node`` command enables you to:
```
Comment (mechanomi) [inline blackbox/docs/admin/cluster-management.rst]: "enables" -> "lets"

File: blackbox/docs/admin/cluster-management.rst [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+ * Unsafely bootstrap a cluster
```
Comment (mechanomi) [inline blackbox/docs/admin/cluster-management.rst]: split infinitive

File: blackbox/docs/admin/cluster-management.rst [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+ * Detach a node from its cluster
+ * Change the role of a node
+ * Eventually recover some data after a disaster
```
Comment (mechanomi) [inline blackbox/docs/admin/cluster-management.rst]: clarify use of "eventually" here

File: blackbox/docs/admin/cluster-management.rst [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+Synopsis
+~~~~~~~~
+
+.. code-block:: console
+
+   bin/crate-node repurpose|detach-cluster|unsafe-bootstrap
+   [--ordinal <Integer>] [-C <KeyValuePair>]
+   [-h, --help] ([-s, --silent] | [-v, --verbose])
+
```
Comment (mechanomi) [inline blackbox/docs/admin/cluster-management.rst]: missing parameters section from original doc

File: blackbox/docs/admin/cluster-management.rst [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+Modes
+~~~~~
+
+This section introduces you to the ``crate-node`` command that is comprised of
```
Comment (mechanomi) [inline blackbox/docs/admin/cluster-management.rst]: first part of this sentence seems redundant

File: blackbox/docs/admin/cluster-management.rst [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+the three modes ``repurpose``, ``unsafe-bootstrap``, and ``detach cluster``.
```
Comment (mechanomi) [inline blackbox/docs/admin/cluster-management.rst]: `detach-cluster`

Comment (mechanomi) [reply blackbox/docs/admin/cluster-management.rst]: these "modes" should be listed in the same order as the list below

File: blackbox/docs/admin/cluster-management.rst [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+These modes allow you to execute all of the above CrateDB cluster management
+operations.
+
+* Use ``crate-node repurpose`` to delete data from a node if it used to be a
```
Comment (mechanomi) [inline blackbox/docs/admin/cluster-management.rst]: perhaps these should be subheadings?

also, perhaps we should give one example use per mode

Comment (kevinkq) [reply blackbox/docs/admin/cluster-management.rst]: All examples and how-tos can be found in the Troubleshooting wit the crate-node CLI section, that we want to link to. The subheading that leads the reader there can be found in the bottom of this doc.

File: blackbox/docs/admin/index.rst [hunk 9]
```diff
@@ -21,6 +21,7 @@ database administrator.
     ssl
     optimization
     jobs-management
+    cluster-management
```
Comment (mechanomi) [inline blackbox/docs/admin/index.rst]: please move this up so that it appears after "Runtime configuration"

File: blackbox/docs/admin/cluster-management.rst [hunk 10]
```diff
@@ -0,0 +1,101 @@
+.. _cluster_management:
+
+==================
+Cluster management
+==================
+
+With the ``crate-node`` CLI tool you can manage your CrateDB clusters by forming
```
Comment (mechanomi) [inline blackbox/docs/admin/cluster-management.rst]: Please reword this so that it mirrors the start of https://crate.io/docs/crate/reference/en/latest/run.html

also, for this intro para, I think there's no need to mention the name of the tool. just something along the lines of this document shows you how to manage nodes in your cluster

File: blackbox/docs/admin/cluster-management.rst [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+new clusters, repairing broken clusters, and by repairing broken nodes.
+
+The ``crate-node`` CLI tool
```
Comment (mechanomi) [inline blackbox/docs/admin/cluster-management.rst]: Change this to "Introduction" like in https://crate.io/docs/crate/reference/en/latest/run.html

File: blackbox/docs/admin/cluster-management.rst [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+---------------------------
+
+The ``crate-node`` CLI tool lets you:
```
Comment (mechanomi) [inline blackbox/docs/admin/cluster-management.rst]: explain where the tool is found and how to run it for the first time. like https://crate.io/docs/crate/reference/en/latest/run.html

perhaps the first run instruction should be the `--help` option

File: blackbox/docs/admin/cluster-management.rst [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+.. NOTE::
+
+  Cluster management operations on a node are only possible while the node is shut down.
+
+Synopsis
+~~~~~~~~
+
+.. code-block:: console
+
+   bin/crate-node repurpose|unsafe-bootstrap|detach-cluster
```
Comment (mechanomi) [inline blackbox/docs/admin/cluster-management.rst]: where is this taken from? the syntax confuses me

File: blackbox/docs/admin/cluster-management.rst [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+   [--ordinal <Integer>] [-C <KeyValuePair>]
+   [-h, --help] ([-s, --silent] | [-v, --verbose])
+
+Modes
```
Comment (mechanomi) [inline blackbox/docs/admin/cluster-management.rst]: rename to "commands"

File: blackbox/docs/admin/cluster-management.rst (hunk 8 above)
Comment (mechanomi) [reply blackbox/docs/admin/cluster-management.rst]: yes. but you only need one example

`crate-node repurpose` is already an example, but it's inline

we should list them as code blocks with `console` formatting and the `sh$` prefix

we should then explain what that command would do

for commands with additional options available to them (indicated in the command synopsis), we should explain that in the relevant subsection and give an example of it being used

File: blackbox/docs/admin/cluster-management.rst [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+  with ``crate-node unsafe-bootstrap``. If ``crate-node unsafe-bootstrap`` was
+  not possible, you can use this mode to move nodes into a brand-new cluster.
+
+.. NOTE::
+
+  The message `Node was successfully detached from the cluster` means that the
+  ``detach-cluster`` tool successfully completed its job. It does not say that there
+  has not been a data loss.
+
+Parameters
```
Comment (mechanomi) [inline blackbox/docs/admin/cluster-management.rst]: please retitle as "Command-line options" per https://crate.io/docs/crate/reference/en/latest/run.html

please also format as a table, per the above document

this section should be moved above the command section

File: blackbox/docs/admin/cluster-management.rst [hunk 16]
```diff
@@ -0,0 +1,101 @@
...
+* ``-h, --help``
+    Return all of the command parameters.
+
+* ``-s, --silent``
+    Show minimal output.
+
+* ``-v, --verbose``
+    Show verbose output.
+
+Examples
```
Comment (mechanomi) [inline blackbox/docs/admin/cluster-management.rst]: this should be reformulated as a "seealso" admonition

File: blackbox/docs/admin/cluster-management.rst [hunk 17]
```diff
@@ -0,0 +1,101 @@
...
+--------
+
+For how-tos and usage examples regarding the ``crate-node`` CLI tool, please
```
Comment (mechanomi) [inline blackbox/docs/admin/cluster-management.rst]: reword this to make it more succinct, once redone as a "seealso" block

File: blackbox/docs/admin/cluster-management.rst (hunk 13 above)
Comment (kevinkq) [reply blackbox/docs/admin/cluster-management.rst]: Here's the source: https://www.elastic.co/guide/en/elasticsearch/reference/current/node-tool.html

File: blackbox/docs/admin/cluster-management.rst [hunk 18]
```diff
@@ -0,0 +1,127 @@
+.. _cluster_management:
+
+==================
+Cluster management
+==================
+
+This document shows you how to manage your CrateDB clusters.
```
Comment (mfussenegger) [inline blackbox/docs/admin/cluster-management.rst]: I think the title, description and placement in the TOC promote this too much as a "general cluster management" tool. 

In day-to-day operations users/admins should almost never have to use this tool. The main use-case is for disaster recovery.

Comment (mechanomi) [reply blackbox/docs/admin/cluster-management.rst]: you're right

perhaps we should make the following changes:

- make a new top-level section in the reference called "CLI tools"
- "CLI tools" should appear in the navigation where "Running CrateDB" does at the moment
- the "Running CrateDB" doc can be moved to "`crate`" under "CLI tools"
- this doc can be moved from "Cluster management" to "`crate-node`" under "CLI tools"

this mirrors the way Elasticsearch do it (see https://www.elastic.co/guide/en/elasticsearch/reference/current/node-tool.html)

Comment (mechanomi) [reply blackbox/docs/admin/cluster-management.rst]: we will need to add in some redirections for this restructuring, but that's fine

Comment (mechanomi) [reply blackbox/docs/admin/cluster-management.rst]: holding off making any other comments for now. I think we should resolve this / perform the changes outlined in my comment above (assuming Jordi agrees) first

Comment (mfussenegger) [reply blackbox/docs/admin/cluster-management.rst]: Not sure if "Running CrateDB" will then still be prominent enough, but otherwise I like the idea.

Comment (kevinkq) [reply blackbox/docs/admin/cluster-management.rst]: Okay, so next I'll execute all of the above. Once done, I'll ping you, Nomi, for the redirects and a review.

File: blackbox/docs/cli-tools.rst [hunk 19]
```diff
@@ -1,9 +1,21 @@
...
+=========
+CLI tools
+=========
+
+.. rubric:: Table of contents
+
+.. contents::
+   :local:
+
+Crate
```
Comment (mechanomi) [inline blackbox/docs/cli-tools.rst]:
```suggestion
``crate``
```

File: blackbox/docs/cli-tools.rst [hunk 20]
```diff
@@ -1,9 +1,21 @@
...
+=====
 
-===============
 Running CrateDB
-===============
+---------------
 
 This document covers the basics of running CrateDB from the command line.
```
Comment (mechanomi) [inline blackbox/docs/cli-tools.rst]: merge this into the following note block then move it after the examples

File: blackbox/docs/cli-tools.rst [hunk 21]
```diff
@@ -16,13 +28,8 @@ This document covers the basics of running CrateDB from the command line.
...
 .. _CrateDB Guide: https://crate.io/docs/crate/guide/en/latest/deployment/index.html
 
-.. rubric:: Table of contents
-
-.. contents::
-   :local:
-
 Introduction
-============
+------------
```
Comment (mechanomi) [inline blackbox/docs/cli-tools.rst]: nix this

File: blackbox/docs/cli-tools.rst [hunk 22]
```diff
@@ -49,7 +56,7 @@ To stop the process that is running in the background send the ``TERM`` or
 The ``crate`` executable supports the following command line options:
```
Comment (mechanomi) [inline blackbox/docs/cli-tools.rst]: nix this para

File: blackbox/docs/cli-tools.rst [hunk 23]
```diff
@@ -49,7 +56,7 @@ To stop the process that is running in the background send the ``TERM`` or
...
 
 Command-line options
```
Comment (mechanomi) [inline blackbox/docs/cli-tools.rst]: nix the example command at the end of this subsection

File: blackbox/docs/cli-tools.rst [hunk 24]
```diff
@@ -101,5 +108,111 @@ The CrateDB process can handle the following signals.
     the node is stopped, you can perform a `graceful stop`_ with the
     :ref:`DECOMMISSION <alter_cluster_decommission>` statement instead.
 
+Crate-node
```
Comment (mechanomi) [inline blackbox/docs/cli-tools.rst]:
```suggestion
``crate-node``
```

File: blackbox/docs/cli-tools.rst [hunk 25]
```diff
@@ -101,5 +108,111 @@ The CrateDB process can handle the following signals.
...
+==========
+
+Managing clusters
+-----------------
+
+This section shows you how to manage CrateDB clusters from the command line.
```
Comment (mechanomi) [inline blackbox/docs/cli-tools.rst]: merge this into the following note block then move it after the example `--help` command

File: blackbox/docs/cli-tools.rst [hunk 26]
```diff
@@ -101,5 +108,111 @@ The CrateDB process can handle the following signals.
...
+
+.. SEEALSO::
+
+   For step-by-step how-tos and examples on how to troubleshoot CrateDB clusters and
+   nodes from the command line, please refer to `Troubleshooting with crate-node CLI`_.
+
+Introduction
```
Comment (mechanomi) [inline blackbox/docs/cli-tools.rst]: nix this heading

File: blackbox/docs/cli-tools.rst [hunk 27]
```diff
@@ -101,5 +108,111 @@ The CrateDB process can handle the following signals.
...
+CrateDB provides a ``crate-node`` command, in the ``bin`` directory, that lets
+you form new clusters, repair broken clusters, and repair broken nodes.
+
+With this command you can:
+
+ * Unsafely bootstrap a cluster
+ * Detach a node from its cluster
+ * Change the role of a node
+
+.. NOTE::
```
Comment (mechanomi) [inline blackbox/docs/cli-tools.rst]: move this to the commands section

also please make sure to indent admonition blocks by 4 spaces

File: blackbox/docs/cli-tools.rst [hunk 28]
```diff
@@ -101,5 +108,111 @@ The CrateDB process can handle the following signals.
...
+
+  Cluster management operations on a node are only possible while the node is
+  shut down.
+
+A good way to start is to invoke ``crate-node`` with the ``-h`` option. This
```
Comment (mechanomi) [inline blackbox/docs/cli-tools.rst]: tighten this into one sentence and try to avoid duplicating literals that appear in the example

File: blackbox/docs/cli-tools.rst [hunk 29]
```diff
@@ -101,5 +108,111 @@ The CrateDB process can handle the following signals.
...
+.. code-block:: console
+
+    sh$ ./bin/crate-node --help
+
+Synopsis
+--------
+
+.. code-block:: console
+
+   bin/crate-node repurpose|unsafe-bootstrap|detach-cluster
```
Comment (mechanomi) [inline blackbox/docs/cli-tools.rst]: needs indenting by 4 chars

start command with `sh$`

if we're going to include this, we should standardize the syntax (preferably using some sort of commonly-used syntax) and also add a synopsis to the `crate` section. you will need an engineer for this. please create an impediment asking for help with this task

File: blackbox/docs/cli-tools.rst [hunk 30]
```diff
@@ -101,5 +108,111 @@ The CrateDB process can handle the following signals.
...
+   [--ordinal <Integer>] [-E <KeyValuePair>]
+   [-h, --help] ([-s, --silent] | [-v, --verbose])
+
+Command-line options
+--------------------
+
+The ``crate-node`` executable supports the following command line options:
+
+
```
Comment (mechanomi) [inline blackbox/docs/cli-tools.rst]: remove duplicate empty new line

Comment (mechanomi) [inline blackbox/docs/cli-tools.rst]: remove duplicate empty new line

File: blackbox/docs/cli-tools.rst [hunk 31]
```diff
@@ -101,5 +108,111 @@ The CrateDB process can handle the following signals.
...
++------------------------+-------------------------------------------------------+
+| ``-s, --silent``       | Show minimal output                                   |
++------------------------+-------------------------------------------------------+
+| ``-v, --verbose``      | Shows verbose output                                  |
++------------------------+-------------------------------------------------------+
+
+Commands
+--------
+
+The ``crate-node`` command provides for the three operations ``repurpose``,
```
Comment (mechanomi) [inline blackbox/docs/cli-tools.rst]: tighten to avoid literals

File: blackbox/docs/cli-tools.rst [hunk 32]
```diff
@@ -101,5 +108,111 @@ The CrateDB process can handle the following signals.
...
+``unsafe-bootstrap``, and ``detach-cluster``. You can use these options to
+repurpose nodes, unsafely bootstrap clusters, and detach nodes from clusters.
+
+Repurpose
```
Comment (mechanomi) [inline blackbox/docs/cli-tools.rst]:
```suggestion
``repurpose``
```

File: blackbox/docs/cli-tools.rst [hunk 33]
```diff
@@ -101,5 +108,111 @@ The CrateDB process can handle the following signals.
...
+^^^^^^^^^
+The ``repurpose`` operation lets you delete data from a node that used to be a
+data node, or a master-eligible node, but has been repurposed to have none of
+these roles.
+
+  .. code-block:: console
+
+      sh$ crate-node repurpose
+
+Unsafe-bootstrap
```
Comment (mechanomi) [inline blackbox/docs/cli-tools.rst]:
```suggestion
``unsafe-bootstrap``
```

File: blackbox/docs/cli-tools.rst [hunk 34]
```diff
@@ -101,5 +108,111 @@ The CrateDB process can handle the following signals.
...
+^^^^^^^^^^^^^^^^
+The ``unsafe-bootstrap`` operation lets you force one of the nodes to form a
```
Comment (mechanomi) [inline blackbox/docs/cli-tools.rst]: missing newline above here

File: blackbox/docs/cli-tools.rst [hunk 35]
```diff
@@ -101,5 +108,111 @@ The CrateDB process can handle the following signals.
...
+new cluster on its own, using its local copy of the cluster metadata. To
+perform unsafe cluster bootstrapping, run:
+
+  .. code-block:: console
+
+      sh$ crate-node unsafe-bootstrap
+
+Detach-cluster
```
Comment (mechanomi) [inline blackbox/docs/cli-tools.rst]:
```suggestion
``detach-cluster``
```

File: blackbox/docs/cli-tools.rst [hunk 36]
```diff
@@ -101,5 +108,111 @@ The CrateDB process can handle the following signals.
...
+Commands
+--------
+
+The ``crate-node`` command provides for the three operations ``repurpose``,
+``unsafe-bootstrap``, and ``detach-cluster``. You can use these options to
+repurpose nodes, unsafely bootstrap clusters, and detach nodes from clusters.
+
+Repurpose
+^^^^^^^^^
+The ``repurpose`` operation lets you delete data from a node that used to be a
```
Comment (mechanomi) [inline blackbox/docs/cli-tools.rst]: "command", not "operation", for consistency. here and elsewhere

File: blackbox/docs/cli-tools.rst [hunk 37]
```diff
@@ -101,5 +108,111 @@ The CrateDB process can handle the following signals.
...
+new cluster on its own, using its local copy of the cluster metadata. To
+perform unsafe cluster bootstrapping, run:
+
+  .. code-block:: console
+
+      sh$ crate-node unsafe-bootstrap
+
+Detach-cluster
+^^^^^^^^^^^^^^
+The ``detach-cluster`` operation lets you move nodes from one cluster to
```
Comment (mechanomi) [inline blackbox/docs/cli-tools.rst]: missing newline

File: blackbox/docs/cli-tools.rst [hunk 38]
```diff
@@ -101,5 +108,111 @@ The CrateDB process can handle the following signals.
...
+another. You can also move nodes into a cluster that you have created using
+``unsafe-bootstrap`` operation. If ``unsafe-bootstrap`` was not possible, it
+also lets you move nodes into a brand-new cluster.
+
+  .. code-block:: console
+
+      sh$ crate-node detach-cluster
+
```
Comment (mechanomi) [inline blackbox/docs/cli-tools.rst]: seperate links at the bottom of the document by two empty lines

File: blackbox/docs/cli-tools.rst (hunk 21 above)
Comment (mechanomi) [inline blackbox/docs/cli-tools.rst]: heading level is wrong (here and elsewhere. check TOC)

File: blackbox/docs/cli-tools.rst [hunk 39]
```diff
@@ -1,9 +1,21 @@
...
+.. rubric:: Table of contents
+
+.. contents::
+   :local:
+
+``crate``
+---------
 
-===============
 Running CrateDB
```
Comment (mechanomi) [inline blackbox/docs/cli-tools.rst]: nix this heading and section

File: blackbox/docs/cli-tools.rst [hunk 40]
```diff
@@ -1,9 +1,21 @@
...
-===============
+---------------
 
 This document covers the basics of running CrateDB from the command line.
```
Comment (mechanomi) [inline blackbox/docs/cli-tools.rst]: move following SEEALSO block to the end of the introductory text

File: blackbox/docs/cli-tools.rst [hunk 41]
```diff
@@ -101,5 +102,116 @@ The CrateDB process can handle the following signals.
     the node is stopped, you can perform a `graceful stop`_ with the
     :ref:`DECOMMISSION <alter_cluster_decommission>` statement instead.
 
+``crate-node``
+--------------
+
+Managing clusters
```
Comment (mechanomi) [inline blackbox/docs/cli-tools.rst]: nix this heading

File: blackbox/docs/cli-tools.rst [hunk 42]
```diff
@@ -101,5 +102,116 @@ The CrateDB process can handle the following signals.
...
+-----------------
+
+This section shows you how to manage CrateDB clusters from the command line.
+
+.. SEEALSO::
```
Comment (mechanomi) [inline blackbox/docs/cli-tools.rst]: move this SEEALSO block to end of intro text

**Comment by mechanomi** [comment]
@kevinkq added some copyedits. can you take a look?

@mkleen @mfussenegger can you take a look at the result of my edits and check for technical accuracy, etc, etc. thx!

File: blackbox/docs/cli-tools.rst [hunk 43]
```diff
@@ -1,209 +1,229 @@
...
+| ``-s, --silent``    | Show minimal output                                 |
++---------------------+-----------------------------------------------------+
+| ``-v, --verbose``   | Shows verbose output                                |
++---------------------+-----------------------------------------------------+
 
-      sh$ crate-node detach-cluster
+.. SEEALSO::
 
-.. _Troubleshooting with crate-node CLI: https://crate.io/docs/crate/guide/en/latest/best-practices/crate-node.html
+    The ``crate-node`` command is modeled on the `elasticsearch-node`_ command.
```
Comment (mkleen) [inline blackbox/docs/cli-tools.rst]: Please don't refer to elastic, we ported over the documentation because we don't want to refer to them.

File: blackbox/docs/cli-tools.rst [hunk 44]
```diff
@@ -1,209 +1,229 @@
...
+    For additional information, consult the Elasticsearch documentation.
 
-.. _Rolling Upgrade: http://crate.io/docs/crate/guide/best_practices/rolling_upgrade.html
 
+.. _deployment guide: https://crate.io/docs/crate/guide/en/latest/deployment/index.html
+.. _Detach a node from its cluster: https://crate.io/docs/crate/guide/en/latest/best-practices/crate-node.html#detach-a-node-from-its-cluster
+.. _elasticsearch-node: https://www.elastic.co/guide/en/elasticsearch/reference/current/node-tool.html
```
Comment (mkleen) [inline blackbox/docs/cli-tools.rst]: Same here, please don't link to elastic.

**Comment by mechanomi** [comment]
@mkleen addressed your comments. can you take another look, please?

**Comment by mechanomi** [comment]
retest this please

**Comment by mechanomi** [comment]
retest this please
